doc/source/install.rst

```diff
@@ -0,0 +1,25 @@
+Installation
+============
+
+.. attention: The nomenclature requires python >= 3.8
+
+
+Via Pip
+-------
+
+.. code-block:: bash
+
+    pip install nomenclature-iamc
+
+.. attention::  The nomenclature package is distributed as "nomenclature-iamc" on pypi.
+
+Installing From Source
+----------------------
+
+nomenclature can also be installed from source. This will get you the latest version
+of the main branch.
+
+.. code-block:: bash
+
+    pip install -e git+https://github.com/IAMconsortium/nomenclature#egg=nomenclature
```

Comment:
I this correct, or is it `pip install git+https://github.com/IAMconsortium/nomenclature.git@main`?

Reply:
Very good question, I am afraid I can only give you what I understand of it which is what I gathered from [this stackoverflow post](https://stackoverflow.com/questions/62841000/how-is-egg-used-in-pip-install-e). 
Basically, if you run `pip install` from a source other than pypi you need to specify the egg option. Apparently this is required otherwise you get an error. I tried this on my machine and can confirm. Interestingly enough it seems that the actual value of the egg parameter does not matter. `pip install -e git+https://github.com/IAMconsortium/nomenclature.git@main#egg=asdf` worked perfectly fine for me. 
To actually answer your question, the `@main` addition is perfectly valid and can be combined with the egg option like so: `pip install -e git+https://github.com/IAMconsortium/nomenclature.git@main#egg=some-name-for-the-egg`. In our case where we install the latest version of the main branch. As this is the default for pip running it with or without the `@main` makes no difference.
This of course raises the question why there is that option in the first place and it is to specify a particular version, so if I actually want to install version 0.1 I would put `@v0.1`.
Hope that clears things up.
I the interest of being precise I'll include the `@main` as it does not rely on perhaps non-obvious default values.

Reply:
Ah got it!

Reply:
The proof of the pip is in the GitHub actions.

In the NetZero2040 project (the next customer for this package, started yesterday), the `requirements.txt` lists
```
git+git://github.com/IAMconsortium/nomenclature.git
```
see [here](https://github.com/netzero2040/netzero2040/blob/5482c3f7bd6456d594aac5f2f10eea433814cb1e/requirements.txt#L3)

and the GitHub Action workflow uses
```
 run: pip install -r requirements.txt
```

see [here](https://github.com/netzero2040/netzero2040/blob/5482c3f7bd6456d594aac5f2f10eea433814cb1e/.github/workflows/validation.yml#L26)

Reply:
I tried to install it on my system and read various different versions of whether or not the egg attribute is needed.
Looking at the [pip documentation](https://pip.pypa.io/en/stable/topics/vcs-support/#url-fragments), suggests that it is.
Weirdly though what you @danielhuppmann posted works perfectly fine.
Only once I started looking at the change log of pip for version [21.2](https://pip.pypa.io/en/stable/news/#id49) I found the following:

> Most of the fragment part, including egg=, is explicitly ignored. Only subdirectory= and hash values (e.g. sha256=) are kept.

So it seems that `git+git://github.com/IAMconsortium/nomenclature.git` will work for pip>=21.2 but will probably throw an error and require an egg for older versions.
My vote would therefore be to keep the egg option in as doesn't hurt in the latest versions of pip but is required in earlier ones.